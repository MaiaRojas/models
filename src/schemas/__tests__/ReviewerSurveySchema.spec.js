const mongoose = require('mongoose/browser');
const { ReviewerSurveySchema, ReviewQuestionSchema } = require('../')(mongoose);

describe('ReviewerSurveySchema', () => {
  it('should fail validation when missing fields are provided', () => {
    const doc = new mongoose.Document({}, ReviewerSurveySchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should fail validation when version not semver compatible', () => {
    const question = new mongoose.Document({}, ReviewQuestionSchema);
    const doc = new mongoose.Document({
      questions: [question._id],
      version: 'foo',
    }, ReviewerSurveySchema);

    return doc.validate()
      .catch(err => expect(err.errors.version.message).toBe('Invalid semver version foo'));
  });

  it('should successfully validate with proper values', (done) => {
    const question = new mongoose.Document({}, ReviewQuestionSchema);
    const doc = new mongoose.Document({
      questions: [question._id],
      version: '2.0.0',
    }, ReviewerSurveySchema);

    return doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
