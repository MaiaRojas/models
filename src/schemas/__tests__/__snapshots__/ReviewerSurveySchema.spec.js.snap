// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewerSurveySchema should fail validation when missing fields are provided 1`] = `
Object {
  "version": [ValidatorError: Path \`version\` is required.],
}
`;
